components/plugin_marketplace/marketplace_list/marketplace_list.js

```diff
@@ -0,0 +1,144 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React from 'react';
+import PropTypes from 'prop-types';
+
+import {FormattedMessage} from 'react-intl';
+
+import {changeOpacity, makeStyleFromTheme} from 'mattermost-redux/utils/theme_utils';
+
+import MarketplaceItem from '../marketplace_item';
+
+const PLUGINS_PER_PAGE = 15;
+
+export default class MarketplaceList extends React.PureComponent {
+    static propTypes = {
+        plugins: PropTypes.array.isRequired,
+        theme: PropTypes.object.isRequired,
+    };
+
+    static getDerivedStateFromProps(props, state) {
+        if (state.page > 0 && props.plugins.length < PLUGINS_PER_PAGE) {
+            return {page: 0};
+        }
+
+        return null;
+    }
+
+    constructor(props) {
+        super(props);
+
+        this.state = {
+            page: 0,
+        };
+    }
+
+    nextPage = (e) => {
+        e.preventDefault();
+        this.setState({page: this.state.page + 1});
+    };
+
+    previousPage = (e) => {
+        e.preventDefault();
+        this.setState({page: this.state.page - 1});
+    };
+
+    renderCount = () => {
+        const total = this.props.plugins.length;
+        const count = this.props.plugins.length;
+        const startCount = this.state.page * PLUGINS_PER_PAGE;
+        const endCount = Math.min(startCount + PLUGINS_PER_PAGE, this.props.plugins.length);
+
+        return (
+            <FormattedMessage
+                id='marketplace_list.countTotalPage'
+                defaultMessage='{startCount, number} - {endCount, number} {count, plural, one {plugin} other {plugins}} of {total, number} total'
+                values={{
+                    count,
```

Comment:
looks like total and count are the same value, can count or total be removed?

Reply:
Done :+1: